frontend/src/DiagramEditor/DiagramEditor.jsx

```diff
@@ -59,18 +78,119 @@ export class DiagramEditor extends React.Component {
     window.removeEventListener("mousemove", this.onWindowMouseMove);
   }
 
+  generateAuthorId = () => {
+    this.authorId = Math.floor(Math.random() * 1000000000000);
+    console.log("authorId:", this.authorId);
+  };
+
+  initialiseWebSocket = () => {
+    const newSocket = new WebSocket(
+      `wss://${WEBSOCKET_API_ID}.execute-api.eu-west-2.amazonaws.com/Prod`
+    );
+
+    this.socket = newSocket;
+
+    // Connection opened
+    newSocket.addEventListener("open", (event) => {
+      console.log("connection open");
+    });
+
+    // Listen for messages
+    newSocket.addEventListener("message", this.onMessageReceived);
+  };
+
+  onMessageReceived = (event) => {
+    const messageData = JSON.parse(event.data);
+    console.log("message:", messageData);
+    switch (messageData.type) {
+      case "master":
+        this.setState({ isMaster: true });
+        break;
+      case "diagramData":
+        this.setState({ diagramData: messageData.diagramData });
+        break;
+      case "connectionId":
```

Comment:
I thought we decided to not send the `connectionId` to the client as we weren't sure whether or not it could be a security issue

Reply:
after discussion, it seems unlikely to be a security issue